### PR TITLE
Update Makefiles in SuiteSparse to make UMFPACK available in python (REVIEW)

### PR DESCRIPTION
**Pull request by sbecuwe**
- include easyconfig for latest SuiteSparse
- add easyconfig for scikit-umfpack
- add patch files to create shared libamd.so and libumfpack.so in SuiteSparse,
  required by scikit-umfpack

requires ~~https://github.com/hpcugent/easybuild-easyblocks/pull/718~~

### CODE REVIEW

File: easybuild/easyconfigs/s/scikit-umfpack/scikit-umfpack-0.2.1-intel-2015b-Python-2.7.10.eb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+pyshortver = '.'.join(pyver.split('.')[:-1])
+sanity_check_paths = {
+    'files': [],
+    'dirs': ['lib/python%s/site-packages/scikits/umfpack' % pyshortver],
+}
+
+moduleclass = 'math'
+
+# Build statistics
+buildstats = [{
```
Comment (boegel) [inline easybuild/easyconfigs/s/scikit-umfpack/scikit-umfpack-0.2.1-intel-2015b-Python-2.7.10.eb]: this shouldn't be here :)

**Comment by boegel** [comment]
Jenkins: ok to test

**Comment by boegel** [comment]
@sbecuwe: missing easyconfigs?

```
EasyBuildError: 'Irresolvable dependencies encountered:
ParMETIS/4.0.3-intel-2015b,
scipy/0.16.0-intel-2015b-Python-2.7.10,
SWIG/3.0.7-intel-2015b-Python-2.7.10'
```

**Comment by boegel** [comment]
@sbecuwe: conflict with `develop` branch resolved with https://github.com/hpcuantwerpen/easybuild-easyconfigs/pull/1

File: easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-4.4.5-intel-2015b-ParMETIS-4.0.3.eb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+toolchain = {'name': 'intel', 'version': '2015b'}
+toolchainopts = {'opt': True, 'unroll': True, 'pic': True}
+
+source_urls = ['http://faculty.cse.tamu.edu/davis/SuiteSparse/']
+sources = [SOURCE_TAR_GZ]
+
+patches = [
+           'AMD_GNUMakefile-4.4.5.patch', 
+           'UMFPACK_GNUMakefile-4.4.5.patch',
+          ]
```
Comment (boegel) [inline easybuild/easyconfigs/s/SuiteSparse/SuiteSparse-4.4.5-intel-2015b-ParMETIS-4.0.3.eb]: please fix indentation, use `%(version)s` and rename the patch files to start with `SuiteSparse-%(version)s`, i.e.:

```python
patches = [
    'SuiteSparse-%(version)s_AMD_GNUMakefile.patch', 
    'SuiteSparse-%(version)s_UMFPACK_GNUMakefile.patch',
]
```

File: easybuild/easyconfigs/s/SuiteSparse/UMFPACK_GNUMakefile-4.4.5.patch [hunk 3]
```diff
@@ -0,0 +1,57 @@
+--- SuiteSparse/UMFPACK/Lib/GNUmakefile.org	2014-03-21 20:16:16.000000000 +0100
```
Comment (boegel) [inline easybuild/easyconfigs/s/SuiteSparse/UMFPACK_GNUMakefile-4.4.5.patch]: please include a comment on top to describe what this patch file does, and include the author

(any lines above this `---` line are comments)

File: easybuild/easyconfigs/s/SuiteSparse/UMFPACK_GNUMakefile-4.4.5.patch [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+ else
+-    I = -I../Include -I../Source -I../../AMD/Include \
+-        -I../../SuiteSparse_config -I../../CHOLMOD/Include
++   I = -I../Include -I../Source -I../../AMD/Include \
++       -I../../SuiteSparse_config -I../../CHOLMOD/Include
+ endif
+ 
++# I = -I../Include -I../Source -I../../AMD/Include -I../../SuiteSparse_config \
++#     -I../../CCOLAMD/Include -I../../CAMD/Include -I../../CHOLMOD/Include \
++#     -I$(EBROOTPARMETIS)/include -I../../COLAMD/Include
```
Comment (boegel) [inline easybuild/easyconfigs/s/SuiteSparse/UMFPACK_GNUMakefile-4.4.5.patch]: this is useless, so remove it (which implies recreating the patch, unless you're very careful in editing the patch :))

File: easybuild/easyconfigs/s/SuiteSparse/UMFPACK_GNUMakefile-4.4.5.patch [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+ #-------------------------------------------------------------------------------
+ # include files:
+@@ -257,8 +261,8 @@
+ 	$(ARCHIVE)  libumfpack.a $^
+ 	- $(RANLIB) libumfpack.a
+ 
+-so: $(II) $(LL) $(GN) $(DI) $(DL) $(ZI) $(ZL)
+-	gcc -shared -Wl,-soname,libumfpack.so -o libumfpack.so $^
++libumfpack.so: $(II) $(LL) $(GN) $(DI) $(DL) $(ZI) $(ZL)
++	$(CC) -shared $^ -lm -L ../../CHOLMOD/Lib -lcholmod -L ../../CAMD/Lib -lcamd -L ../../COLAMD/Lib -lcolamd $(EBROOTPARMETIS)/lib/libparmetis.a $(EBROOTPARMETIS)/lib/libmetis.a -L../../CCOLAMD/Lib -lccolamd -L ../../SuiteSparse_config -lsuitesparseconfig ../../AMD/Lib/libamd.so -Wl,-soname -Wl,$@ -o libumfpack.so
```
Comment (boegel) [inline easybuild/easyconfigs/s/SuiteSparse/UMFPACK_GNUMakefile-4.4.5.patch]: avoid long line?

File: easybuild/easyconfigs/s/SuiteSparse/AMD_GNUMakefile-4.4.5.patch [hunk 6]
```diff
@@ -0,0 +1,42 @@
+--- SuiteSparse/AMD/Lib/GNUmakefile.org	2014-03-21 20:22:00.000000000 +0100
```
Comment (boegel) [inline easybuild/easyconfigs/s/SuiteSparse/AMD_GNUMakefile-4.4.5.patch]: please include a comment at the top describing what this patch does, and who the author is

**Comment by sbecuwe** [comment]
Pull request currently on hold due to a (possible dependency) problem identified by a user.

**Comment by boegel** [comment]
@sbecuwe: please merge https://github.com/hpcuantwerpen/easybuild-easyconfigs/pull/2 to fix a tiny style remark in this PR?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/e491b0ad2bc2ab5cc52a for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in this PR)
Linux centos linux 7.1.1503, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/4ddb72703d43f098e047 for a full test report.

**Comment by boegel** [comment]
good to go in, tested on top of https://github.com/hpcugent/easybuild-easyblocks/pull/718

thanks @sbecuwe!

**Pull request merged**